### ISSUE TITLE
[C&P] Introduce the first `session.feature`

### ISSUE THREAD

**Issue by Olshansk** [pokt-network/poktroll#232]
## Objective

Add the first C&P E2E test to create a foundation for future ones.

## Origin Document

Having enabled the E2E relay in CI, this is the next logical step to finish off the MVP business logic.

## Goals

- Identify missing gaps in tooling to implement an E2E C&P lifecycle
- Set the standard in how E2E C&P lifecycle tests should be handeled

## Deliverables

- [x] Add a `session.feature` file
- [ ] Implement AT LEAST 1 test that validates AT LEAST 1 end-to-end claim & proof lifecycle verifying the:
	- [ ] Non-existence of on-chain claims & proofs
	- [ ] The persistence of a proof after session end
	- [x] The persistence of a claim after a sufficient time has passed
	- [ ] Changes in account balances (application & supplier) after the proof is validated
- [ ] Add an assertion on the sum of the tree in E2E Tests (final 8 bits is the value)

## Non-goals / Non-deliverables

- Exhaustively accounting for all success & failure cases of the C&P lifecycle
- Implementing gaps in the C&P on-chain business logic

## General deliverables

- [ ] **Comments**: Add/update TODOs and comments alongside the source code so it is easier to follow.
- [ ] **Testing**: Add new tests (unit and/or E2E) to the test suite.
- [ ] **Makefile**: Add new targets to the Makefile to make the new functionality easier to use.
- [ ] **Documentation**: Update architectural or development READMEs; use [mermaid](https://mermaid-js.github.io/mermaid/) diagrams where appropriate.

---

**Creator**: @Olshansk 
**Co-Owners**: @bryanchriswhite

**Comment by Olshansk** [comment]
@bryanchriswhite I added the following deliverable:
```
- [ ] Add an assertion on the sum of the tree in E2E Tests (final 8 bits is the value)
```

**Comment by bryanchriswhite** [comment]
Closed by #328
